src/Utilities/ErrorHandling/AbortWithErrorMessage.cpp

```diff
@@ -20,13 +20,14 @@ struct FillBacktrace {};
 // Convert the address to the virtual memory address inside the
 // library/executable. This is the address that addr2line and llvm-addr2line
 // expect.
-size_t convert_to_virtual_memory_address(const void* addr) {
+void* convert_to_virtual_memory_address(const void* addr) {
   Dl_info info;
   link_map* link_map = nullptr;
   dladdr1(addr, &info,
           reinterpret_cast<void**>(&link_map),  // NOLINT
           RTLD_DL_LINKMAP);
-  return reinterpret_cast<size_t>(addr) - link_map->l_addr;
+  return reinterpret_cast<void*>(reinterpret_cast<size_t>(addr) -
+                                 link_map->l_addr);
```

Comment:
Sorry, one more finicky cast thing - can you cast the `link_map->l_addr` to `ptrdiff_t`?

Reply:
You are not allowed to do math on a `void*`, gives compiler error

Reply:
ah, sure, now that you say that it makes sense. It couldn't possibly know the stride. Good enough then :shrug: